### ISSUE TITLE
Plugin seems to wait until webpack is done being built

### ISSUE THREAD

**Issue by justinmwarner** [TypeStrong/fork-ts-checker-webpack-plugin#159]
Hello,

We've been using this plugin + transpileOnly for a while now. Previously, it brought our build times incremental down from 25 sec. to 6 sec., which was awesome.

However, sometime in the past month, it seems to have slowed down again.

Judging by the output, it seems this plugin is working successfully, however it waits until Webpack is finished building it (which is ~ 35 seconds) until it allows the page to refresh.

Please take a look at this output, which I think shows the above:
i ｢wdm｣:
i ｢wdm｣: Compiled successfully.
i ｢wdm｣: Compiling...
webpack building...
No type errors found
Version: typescript 3.0.1
Time: 8857ms
webpack built fa569d26026fe1d4035f in 35845ms

When the 8857ms message is printed, and the notification system toast comes up, I try refreshing the page, and it does nothing until the next line, webpack built is printed.

Any reason this may occur? Am I reading the above correctly? Sorry for the vague explanation, I'm not too sure where to start on debugging this.

Thanks!

**Comment by johnnyreilly** [comment]
What changed? Did you move to a new version that caused a performance regression? Or has your config changed?

**Comment by johnnyreilly** [comment]
Did your config change? Is `async` false?

**Comment by justinmwarner** [comment]
Hey John.  Yes, we did update to new versions, but even after I revert that, it doesn't seem to fix the problem.

I've tried with async set to true and false. But in the past, it's always been the default, we never specified it.

Is there something I can do to debug this on my side?

To rule out machine specific, it's happening on a co-workers machine as well.

**Comment by johnnyreilly** [comment]
`async: true` is the default and it's what you likely want for development. It's the setting that says "hey I'm not waiting for webpack!"

The fact that you've rolled back to an older version without resolving the issue makes me think that it's not a problem with the plugin. (Which is a relief for me although it doesn't resolve your issue.)

I suspect this is a config issue or some curiosity introduced to the codebase. If I was you I'd use `git checkout ...` to roll back to older versions of your codebase until you identify the commit that introduced this issue. As described here:

https://stackoverflow.com/a/2007704/761388

Not a fun job but should get you the information you desire. Good luck!

**Comment by justinmwarner** [comment]
Hey John,  Thanks for the reply.  So, I've narrowed down the commit that broke this.  Essentially, we added a single json file that we use in both SASS and Typescript. So we added a library to deal with this, https://github.com/Updater/node-sass-json-importer.

I have two theories on what's happening:

1. Updating any code forces all scss to be re-built, adding that time to the overall time, thus the bundle isn't ready yet.  Someone posted this on the following issue: https://github.com/Updater/node-sass-json-importer/issues/15

2. Because ts-loader now relies on the json file to be done, ts-loader, thus this plugin, is waiting until the json file work is done before proceeding, making a dependency. This semi-relates to 1 above, but I think is different.

I'm closing this though, as this is def. not on this library. Thanks for the prompt replies.
